src/coordinatorapi.cpp

```diff
@@ -809,6 +809,43 @@ sendQueryToCoordinator(const char *id,
   return *val_len;
 }
 
+int
+sendQueryAllToCoordinator(const char *id, void **buf)
+{
+  DmtcpMessage msg(DMT_NAME_SERVICE_QUERY_ALL);
+
+  JWARNING(strlen(id) < sizeof(msg.nsid));
+  strncpy(msg.nsid, id, sizeof msg.nsid);
+  int sock = coordinatorSocket;
+  if (dmtcp_is_running_state()) {
+    if (nsSock == -1) {
+      nsSock = createNewSocketToCoordinator(COORD_ANY);
+      JASSERT(nsSock != -1);
+      nsSock = Util::changeFd(nsSock, PROTECTED_NS_FD);
+      JASSERT(nsSock == PROTECTED_NS_FD);
+      DmtcpMessage m(DMT_NAME_SERVICE_WORKER);
+      JASSERT(Util::writeAll(nsSock, &m, sizeof(m)) == sizeof(m));
+    }
+    sock = nsSock;
+  }
+
+  JASSERT(Util::writeAll(sock, &msg, sizeof(msg)) == sizeof(msg));
+  msg.poison();
+
+  JASSERT(Util::readAll(sock, &msg, sizeof(msg)) == sizeof(msg));
+  msg.assertValid();
+
+  JASSERT(msg.type == DMT_NAME_SERVICE_QUERY_ALL_RESPONSE &&
+          msg.extraBytes == msg.valLen);
+
+  if (buf == NULL) {
```

Comment:
I think you mean `*buf`, instead of `buf` in the condition statement. `buf` in this context is the address of the user buffer, and I don't think it should ever be NULL, even if the user buffer is.